File: solutions/c/rna-transcription/1/rna_transcription.c
#include "rna_transcription.h"

#include <stdlib.h>
#include <stddef.h>
#include <string.h>

char *to_rna(const char *dna) {
    size_t length = strlen(dna);
    char *rna = malloc(sizeof(char) * (length + 1));
    rna[length] = '\0';

    for (size_t i = 0; i < length; i++) {
        if (dna[i] == 'G') {
            rna[i] = 'C';
        } else if (dna[i] == 'C') {
            rna[i] = 'G';
        } else if (dna[i] == 'T') {
            rna[i] = 'A';
        } else {
            rna[i] = 'U';
        }
    }

    return rna;
}

File: solutions/go/rna-transcription/1/rna_transcription.go
// Package strand provides a function for transcribing RNA.
package strand

var toRnaMap = map[rune]string{
	'G': "C", 'C': "G", 'T': "A", 'A': "U",
}

// ToRNA returns the RNA complement string of the given DNA sequence string.
func ToRNA(dna string) (rna string) {
	for _, c := range dna {
		rna += toRnaMap[c]
	}
	return
}
